contracts/DaiHrd.sol

```diff
@@ -486,38 +479,38 @@ contract DaiHrd is ERC777, MakerFunctions {
 		}
 	}
 
-	function deposit(uint256 attodai) external returns(uint256 depositedAttopot) {
+	function deposit(uint256 attodai) external returns(uint256 attodaiHrd) {
 		dai.transferFrom(msg.sender, address(this), attodai);
 		daiJoin.join(address(this), dai.balanceOf(address(this)));
-		depositedAttopot = depositVatDaiForAccount(msg.sender);
+		uint256 depositedAttopot = depositVatDaiForAccount(msg.sender);
 		emit Deposit(msg.sender, attodai, depositedAttopot);
 		return depositedAttopot;
 	}
 
-	// If the user has vat dai directly (after performing vault actions, for instance), they don't need to create the DAI ERC20 for us to burn it, we'll accept vat dai
-	function depositVatDai(uint256 attoVatDai) external returns(uint256 depositedAttopot) {
-		uint256 attorontodai = attoVatDai.mul(10 ** 27);
-		vat.move(msg.sender, address(this), attorontodai);
-		depositedAttopot = depositVatDaiForAccount(msg.sender);
-		emit DepositVatDai(msg.sender, attorontodai, depositedAttopot);
+	// If the user has vat dai directly (after performing vault actions, for instance), they don't need to create the DAI ERC20 just so we can burn it, we'll accept vat dai
+	function depositVatDai(uint256 attorontovatDai) external returns(uint256 attodaiHrd) {
+		vat.move(msg.sender, address(this), attorontovatDai);
+		uint256 depositedAttopot = depositVatDaiForAccount(msg.sender);
+		emit DepositVatDai(msg.sender, attorontovatDai, depositedAttopot);
 		return depositedAttopot;
 	}
 
 	function withdrawTo(address recipient, uint256 attodaiHrd) external returns(uint256 attodai) {
-		pot.drip();
-		return _withdraw(recipient, attodaiHrd);
+		// Don't need rontodaiPerPot, so we don't call updateAndFetchChi
+		if (pot.rho() != now) pot.drip();
+		return withdraw(recipient, attodaiHrd);
 	}
 
 	function withdrawToDenominatedInDai(address recipient, uint256 attodai) external returns(uint256 attodaiHrd) {
 		uint256 rontodaiPerPot = updateAndFetchChi();
 		attodaiHrd = convertAttodaiToAttodaiHrd(attodai, rontodaiPerPot);
-		uint256 attodaiWithdrawn = _withdraw(recipient, attodaiHrd);
-		require(attodaiWithdrawn >= attodai, "Not withdrawing enough DAI to cover request");
+		uint256 attodaiWithdrawn = withdraw(recipient, attodaiHrd);
+		require(attodaiWithdrawn >= attodai, "DaiHrd/withdrawToDenominatedInDai: Not withdrawing enough DAI to cover request");
```

Comment:
Beefed up some of our require messages.  Though, in hindsight, is there a problem with require messages greater than 32 bytes long?  I don't think so... but didn't test either.  😬

Reply:
(it appears this message was already over 32 bytes long, so if there is a problem we already had it)